### PR TITLE
Change New-ModuleManifest encoding to UTF8NoBOM on non-Windows platforms

### PR DESCRIPTION
**Pull request by anmenaga**
This addresses #3789 
Currently New-ModuleManifest creates psd1 manifests in UTF-16 with BOM; this creates problems for Linux tools and Git diff'er.
This fix changes encoding of New-ModuleManifest to be UTF8 (no BOM).
Also mentioned New-ModuleManifest in "[RFC 0020 DefaultFileEncoding](https://github.com/PowerShell/PowerShell-RFC/blob/master/1-Draft/RFC0020-DefaultFileEncoding.md)".

### CODE REVIEW

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs
Comment (Francisco-Gamino) [inline src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: Do we need to update the documentation for this change?

**Comment by joeyaiello** [comment]
As it currently stands, this is a breaking change (appropriate label added). 

Based on our current POR, we're okay changing the default *on non-Windows platforms*, but this shouldn't be changed (yet) on Windows until we've had the breaking change conversation. /cc @JamesWTruher

**Comment by joeyaiello** [comment]
I understand that this currently breaks the `git` scenario, we just have to be thoughtful about this.

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs
Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: @anmenaga says he opened a doc issue, can you link it here?

**Comment by vors** [comment]
Are there plans to add UTF8-NoBom to powershell supported Encodings?
I.e. for `Set-Content` and friends.

Changing the most important scenarios is good, but it would be great to address the problem with encodings broader and consistently to help shape the conversation about breaking changes.

**Comment by iSazonov** [comment]
@vors UTF8noBOM is discussed in [RFC 0020 DefaultFileEncoding](https://github.com/PowerShell/PowerShell-RFC/blob/master/1-Draft/RFC0020-DefaultFileEncoding.md).

**Comment by iSazonov** [comment]
New-PSSessionConfigurationFile uses `EncodingConversion.Unicode` too.

**Comment by joeyaiello** [comment]
@vors: first, yes. We're adding that (and making it the default across the board on Linux). The hard part is what to do with Windows (change the defaults? leave a legacy enum value for the "traditional" disparate defaults?) Which is why I want to have the discussion.

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs
Comment (anmenaga) [inline src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: [This PR](https://github.com/PowerShell/PowerShell-RFC/pull/95) in RFC repo.

**Comment by anmenaga** [comment]
@joeyaiello good point;
PR updated to change encoding only for non-Windows.

**Comment by SteveL-MSFT** [comment]
Is there an actual problem on Windows using UTF8NoBOM?  If only English is used, it's equivalent to an ascii file and Windows tools handle that just fine and I expect that to be the majority of use.

**Comment by TravisEz13** [comment]
Restarted AppVeyor due to MyGet connectivity issue

**Comment by joeyaiello** [comment]
@anmenaga if you could sync with @JamesWTruher before merging this, that'd be awesome. Assuming the @PowerShell/powershell-committee votes to accept tomorrow, we want to move to UTF8NoBOM everywhere.

**Comment by iSazonov** [comment]
@SteveL-MSFT 
>Is there an actual problem on Windows using UTF8NoBOM? 

Users can add non-English comments in the manifest.

**Comment by SteveL-MSFT** [comment]
@iSazonov absolutely, but I believe the majority of usage is English.  Even with non-ascii characters, my understanding is that notepad has heuristics to determine encoding to display it correctly.  VSCode seems to support UTF8NoBom just fine.

**Comment by iSazonov** [comment]
>notepad has heuristics to determine encoding to display it correctly

It seem works on last versions only not on Windows 7/8(?).

**Comment by TravisEz13** [comment]
@joeyaiello  Are you saying this is blocked on a sign off from @JamesWTruher ?

**Comment by TravisEz13** [comment]
@joeyaiello @JamesWTruher  Do we want to document the output format of Cmdlets?

**Comment by joeyaiello** [comment]
@TravisEz13 yes and yes. I think the uber-point is that @JamesWTruher is doing so much work here, that it might get changed anyway (and it might just muddle his extensive notes). 

Ultimately up to him, though. I just wanted to make sure everyone was on the same page. 

Regarding docs though: heck yes. We'll have a `Legacy` value for Encoding, and that will be well-documented.

**Comment by JamesWTruher** [comment]
This is fine for now, I'll be changing all the cmdlets in response to https://github.com/PowerShell/PowerShell-RFC/blob/master/1-Draft/RFC0020-DefaultFileEncoding.md.

**Review by JamesWTruher — APPROVED** [review]
This is fine for now, I'll be changing all the cmdlets which provide for encoding as part of https://github.com/PowerShell/PowerShell-RFC/blob/master/1-Draft/RFC0020-DefaultFileEncoding.md. This doesn't muddy the waters for me.

File: test/powershell/engine/Module/NewModuleManifest.Tests.ps1
Comment (TravisEz13) [inline test/powershell/engine/Module/NewModuleManifest.Tests.ps1]: Could you just calculate the expected value in a `beforeall` then have one test for all platforms?

**Review by TravisEz13 — COMMENTED** [review]
Please respond to my comment on your code.

**Comment by TravisEz13** [comment]
@anmenaga Also, as a PowerShell team member, you need to submit the PR to update the docs before I merge this.  Please add a link to the PR here.  Thanks.

**Comment by iSazonov** [comment]
For encoding changes, you might want to delay documenting as we expect big changes in this.

File: test/powershell/engine/Module/NewModuleManifest.Tests.ps1
Comment (anmenaga) [inline test/powershell/engine/Module/NewModuleManifest.Tests.ps1]: Good idea; updated.

**Comment by anmenaga** [comment]
@TravisEz13
1) [This PR](https://github.com/PowerShell/PowerShell-RFC/pull/95) in RFC repo was merged;
2) I've ran a search in PowerShell-Docs and didn't find any mentioning of encoding for New-ModuleManifest, so there is nothing to update there. Besides, this PR will be anyway overwritten by Jim's encoding work that is arriving soon.

**Review by SteveL-MSFT — DISMISSED** [review]
We should update https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md with the recommended settings in VSCode for automatic removing of trailing whitespace for resx files

**Comment by TravisEz13** [comment]
@anmenaga Per the conversation in this PR, we want to add encoding documentation to the CmdLet docs.

**Comment by anmenaga** [comment]
@TravisEz13 Here is the doc PR: https://github.com/PowerShell/PowerShell-Docs/pull/1405

**Comment by kilasuit** [comment]
My only issue here is that although this is a change that I agree with, I think that this is where I have concerns of this causing a disparate and inconsistent behaviour between all versions of PowerShell below v6 due to the fact that New-ModuleManifest is baked into the **Microsoft.PowerShell.Core** module.

I personally would **much** prefer to see that this update to the cmdlet gets released in a manner that **_fully_** supports being used on downlevel systems where PowerShell v6 is unable to be installed.

However the only real solutions that I can really see available to be able to make that to happen would be to 
- Investigate the decoupling of a number of the cmdlets from the **Microsoft.PowerShell.Core** module into their own separate individual modules or a condensed updated singular module that has all the updates to any module in **Microsoft.PowerShell.Core** (not an overly fantastic solution) - similar in the idea put forward in #1979 
- Look to recreate this cmdlet within the PowerShellGet module.  
- Alongside the v6 updates look to release a completely updated and downlevel compatible version of the **Microsoft.PowerShell.Core** module to the PowerShell Gallery, which isn't really an overly viable solution either.

If this change could be easily replicated for downlevel versions of PowerShell I think that would be the best outcome all around.

**Comment by TravisEz13** [comment]
@kilasuit Thanks for the feedback.  I think most of the concerns raised would be best tracked in #1979 or a new issue.

**Pull request merged**
